src/Cardano/Address/Style/Jormungandr.hs

```diff
@@ -0,0 +1,313 @@
+
+    deriveAddressPrivateKey (Jormungandr accXPrv) accountingStyle addrIx =
+        let
+            changeCode =
+                toEnum @(Index 'Soft _) $ fromEnum accountingStyle
+            changeXPrv = -- lvl4 derivation; soft derivation of change chain
+                deriveXPrv DerivationScheme2 accXPrv changeCode
+            addrXPrv = -- lvl5 derivation; soft derivation of address index
+                deriveXPrv DerivationScheme2 changeXPrv addrIx
+        in
+            Jormungandr addrXPrv
+
+instance SoftDerivation Jormungandr where
+    deriveAddressPublicKey (Jormungandr accXPub) accountingStyle addrIx =
+        fromMaybe errWrongIndex $ do
+            let changeCode = toEnum @(Index 'Soft _) $ fromEnum accountingStyle
+            changeXPub <- -- lvl4 derivation in bip44 is derivation of change chain
+                deriveXPub DerivationScheme2 accXPub changeCode
+            addrXPub <- -- lvl5 derivation in bip44 is derivation of address chain
+                deriveXPub DerivationScheme2 changeXPub addrIx
+            return $ Jormungandr addrXPub
+      where
+        errWrongIndex = error $
+            "deriveAddressPublicKey failed: was given an hardened (or too big) \
+            \index for soft path derivation ( " ++ show addrIx ++ "). This is \
+            \either a programmer error, or, we may have reached the maximum \
+            \number of addresses for a given wallet."
+
+instance StakingDerivation Jormungandr where
+    deriveStakingPrivateKey (Jormungandr accXPrv) =
+        let
+            changeXPrv = -- lvl4 derivation; soft derivation of change chain
+                deriveXPrv DerivationScheme2 accXPrv (toEnum @(Index 'Soft _) 2)
+            stakeXPrv = -- lvl5 derivation; soft derivation of address index
+                deriveXPrv DerivationScheme2 changeXPrv (minBound @(Index 'Soft _))
+        in
+            Jormungandr stakeXPrv
+
+-- | 'NetworkDiscriminant' for Cardano Incentivized Testnet testnet & Jormungandr
+--
+-- @since 2.0.0
+incentivizedTestnet :: NetworkDiscriminant Jormungandr
+incentivizedTestnet = NetworkTag 0x83
+
+instance HasNetworkDiscriminant Jormungandr where
+    type NetworkDiscriminant Jormungandr = NetworkTag
+    addressDiscrimination _ = RequiresNetworkTag
+    networkTag = id
+
+instance PaymentAddress Jormungandr where
+    paymentAddress discrimination k = unsafeMkAddress $
+        invariantSize expectedLength $ BL.toStrict $ runPut $ do
+            putWord8 (invariantNetworkTag 255 firstByte)
+            putByteString (getPublicKey $ getKey k)
+      where
+          firstByte = networkTag @Jormungandr discrimination
+          expectedLength = 1 + publicKeySize
+
+instance DelegationAddress Jormungandr where
+    delegationAddress discrimination paymentKey stakingKey = unsafeMkAddress $
+        invariantSize expectedLength $ BL.toStrict $ runPut $ do
+            putWord8 (invariantNetworkTag 255 $ NetworkTag $ firstByte + 1)
```

Comment:
`+1` :thinking:  ?

Reply:
for paymentAddress singleAddress is used, for delegationAddress - groupedAddress : https://github.com/input-output-hk/cardano-wallet/blob/master/lib/core/src/Cardano/Wallet/Primitive/AddressDerivation/Shelley.hs#L343-L344 - if I understand well

Reply:
:+1: